managed/resource_cluster.go

```diff
@@ -916,6 +1154,63 @@ func resourceClusterRead(ctx context.Context, accountId string, projectId string
 		}
 	}
 
+	cmkResp, _, err := apiClient.ClusterApi.GetClusterCMK(context.Background(), accountId, projectId, clusterId).Execute()
+	if cmkResp.Data.Get() != nil {
+		cmkSpec := CMKSpec{}
+		cmkData := cmkResp.GetData()
+
+		cmkSpec.ProviderType = types.String{Value: string(cmkData.GetProviderType())}
+		cmkSpec.IsEnabled = types.Bool{Value: bool(cmkData.GetIsEnabled())}
+
+		switch cmkSpec.ProviderType.Value {
```

Comment:
Can common code be extracted out of this switch case and the switch case statement above?

Reply:
The above switch case is building a openapiclient.CMKSpec, however this is building the go model CMKSpec.
Can you please tell how to common out the code in both the cases?